Add thank you screen for akismet checkout

CodeyGuyDylan opened the pull request:
## Proposed Changes

This PR adds a thank you screen when you get finished buying an Akismet product at the route `/checkout/akismet/thank-you/:productSlug`

Worked with @ballio2000 on this design

Note: We plan to show the API key here in the future

## Testing Instructions

1. Checkout this branch
2. Sandbox `public-api.wordpress.com`
3. In your 0-sandbox.php file, add the following line
```php
define( 'USE_STORE_SANDBOX', true );
```
4. Start your local environment via `yarn start`
5. Head to http://calypso.localhost:3000/checkout/akismet/ak_plus_yearly_3
6. Make sure the styles look good on desktop and mobile
![image](https://user-images.githubusercontent.com/65001528/226464081-1c367abd-c848-461b-81f6-7e1b09ccdf39.png)
![image](https://user-images.githubusercontent.com/65001528/226464236-612c11cf-d9a1-429a-8684-52d41ef41867.png)

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Commit:
Add thank you screen for akismet checkout

CodeyGuyDylan commented on client/lib/analytics/ad-tracking/constants.js:
```diff
@@ -63,6 +63,7 @@ export const TRACKING_IDS = {
 	jetpackGoogleAnalyticsGtag: 'UA-52447-43', // Jetpack Gtag (Analytics) for use in Jetpack x WordPress.com Flows
 	jetpackGoogleGA4Gtag: 'G-YELRMVV4YG',
 	jetpackGoogleAdsGtagPurchase: 'AW-946162814/kIF1CL3ApfsBEP6YlcMD',
+	akismetGoogleGA4Gtag: 'G-V8X5PZE9F8',
```
Needed to add the GA4 tag for Akismet in order to add page view tracking to this page

robertsreberski replied:
If you feel that this comment was necessary for reviewers - maybe it's worth to add it within the code as well, so that future collaborators will also understand that decision?

CodeyGuyDylan replied:
I am actually not sure that is the case, the main reason I felt it relevant to the reviewers was in case they were wondering why I was touching so many analytics files when this is a task not specific to analytics. Do you think it is necessary for the code?

CodeyGuyDylan commented on client/my-sites/checkout/checkout-thank-you/style.scss:
```diff
@@ -833,6 +833,71 @@
+	}
+
+	.akismet-checkout-thank-you__emoji {
+		font-size: 40px; /* stylelint-disable-line declaration-property-unit-allowed-list */
```
Explanation for using an ignore comment: Designer wanted this to be 40px, and there is no way with the allowed `rem` values or with the existing variables that I am aware of to get this value without hard coding. I tried using 2.5rem but I got a lint error saying to use one of the allowed values, and there was 2.25 and 2.75 but nothing in between.

jboland88 replied:
thanks for the explainer! Seems OK as a one-off. If we need to use that size more commonly, seems like it would be OK to add to the lint config.

jboland88 commented on client/my-sites/checkout/index.js:
```diff
@@ -107,6 +108,17 @@ export default function () {
 		);
 	}
 
+	if ( isEnabled( 'akismet/siteless-checkout' ) ) {
```
Nitpick - we could likely group this `page()` in with the existing conditional.

jboland88 commented on client/my-sites/checkout/index.js:
```diff
@@ -107,6 +108,17 @@ export default function () {
+		);
+	}
+
 	// TODO: need thank you page for akismet
```
not anymore ;)

jboland88 commented on client/my-sites/checkout/checkout-thank-you/akismet-checkout-thank-you.tsx:
```diff
@@ -0,0 +1,71 @@
+			/>
+
+			<Card className="akismet-checkout-thank-you__card">
+				{ /* An improvment here would be to add a type filter on wpcom like jetpack has. See: 2f832-pb/ */ }
```
It took me a moment to understand what this comment was referencing - is there a way we can make it more clear?

jboland88 replied:
But yes, I agree that filtering this query would speed things up :)

CodeyGuyDylan replied:
Sure I'll clear up that comment, I'll add it as a NTH task on the board

jboland88 reviewed (approved):
This tested well. I have some questions/ suggestions that we can handle in a separate PR(s).

Can we show a button of some kind to link to the purchase in Calypso? Maybe a "Mange Purchase" button?

The product name loading in the thank you screen is a bit slow, agreed that filtering products in `<QueryProudcts>` could help with that.

I don't know much about the analytics implementation overall, but this change looks sensible.

andrii-lysenko reviewed (approved):
The code looks neat and clean and the logic behaves as described. There are 2 things that we probably could add, though, it can be done as a follow-up.
- We could probably get some `license key` or some other information retrieved by `receipt_id` as we do when purchasing Jetpack products so there would be something user could stick to.
- We also probably could have some `Continue` button which navigates to Akismet docs or something like that.

These are optional and definitely could be addressed in the future, while this PR is good to go as is 👍 .

Commit:
Group feature flagged pages together

Commit:
Remove todo comment

Commit:
Clear up comment about slow products query

CodeyGuyDylan commented:
> The code looks neat and clean and the logic behaves as described. There are 2 things that we probably could add, though, it can be done as a follow-up.
> 
> * We could probably get some `license key` or some other information retrieved by `receipt_id` as we do when purchasing Jetpack products so there would be something user could stick to.
> * We also probably could have some `Continue` button which navigates to Akismet docs or something like that.
> 
> These are optional and definitely could be addressed in the future, while this PR is good to go as is 👍 .

I agree, I will add these as tasks to the board

CodeyGuyDylan commented:
> Can we show a button of some kind to link to the purchase in Calypso? Maybe a "Mange Purchase" button?

Added a task to add a CTA

> The product name loading in the thank you screen is a bit slow, agreed that filtering products in `<QueryProudcts>` could help with that.

Added a task for this

> I don't know much about the analytics implementation overall, but this change looks sensible.

I consulted @robertsreberski  on this change, he said he'd look over it when it was ready

robertsreberski commented on client/my-sites/checkout/checkout-thank-you/style.scss:
```diff
@@ -833,6 +833,71 @@
+		background-size: cover;
+
+		width: 100%;
+		height: 100vw;
```
Do I understand correctly that we try to set `full viewport width` to `height` property of this element?

CodeyGuyDylan replied:
Yes that is correct, it seems to work pretty well though I am aware it is a bit of a hack

robertsreberski commented on client/lib/analytics/ad-tracking/google-analytics.js:
```diff
@@ -38,15 +38,25 @@ export function getGoogleAnalyticsDefaultConfig() {
 	urlPath,
 	pageTitle,
-	useJetpackGoogleAnalytics = false
+	useJetpackGoogleAnalytics = false,
```
as we grow our list of exceptions, we could consider creating an object with `use(...)GoogleAnalytics`? I can see that it could be out of the scope for that PR - but maybe let's create a followup maintenance task to address it later?

CodeyGuyDylan replied:
Yes I agree with this. I won't hold up this PR going out, I'll add it to maintenance

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7880634

Thank you @CodeyGuyDylan for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7880634).
